deps/rabbitmq_stream/src/rabbit_stream_reader.erl

```diff
@@ -1923,19 +1932,25 @@ handle_frame_post_auth(Transport,
                        {credit, SubscriptionId, Credit}) ->
     case Consumers of
         #{SubscriptionId := Consumer} ->
-            #consumer{credit = AvailableCredit} = Consumer,
+            #consumer{credit = AvailableCredit, last_listener_offset = LLO} =
+                Consumer,
             case send_chunks(Transport,
                              Consumer,
                              AvailableCredit + Credit,
+                             LLO,
                              SendFileOct)
             of
                 {error, closed} ->
                     rabbit_log_connection:info("Stream protocol connection has been closed by "
                                                "peer",
                                                []),
                     throw({stop, normal});
-                {{segment, Log1}, {credit, Credit1}} ->
-                    Consumer1 = Consumer#consumer{log = Log1, credit = Credit1},
+                {{segment, Log1}, {credit, Credit1},
```

Comment:
`send_chunks` takes a Consumer as an argument, why not just modify it there and return the updated consumer?